Tests/FunctionalTests/MiscellaneousTests.swift

```diff
@@ -435,4 +436,44 @@ class MiscellaneousTestCase: XCTestCase {
         }
       #endif
     }
+
+    func testDynamicallyLinkedExecutable() {
```

Comment:
I think testing the flags in build plan is good enough. We usually avoid costly integration tests in the main repository unless they're going to provide a lot of value. We can move it to https://github.com/apple/swift-integration-tests repository or try to consolidate it with an existing dylib integration test to minimize the cost.

Reply:
I removed the weighty test.

Instead I added an single `XCTAssert` after one of the flag specs that checks for these specific flags and if they disappear, its failure message describes why they are necessary.